packages/interactivity/src/proxies/test/state-proxy.ts

```diff
@@ -0,0 +1,1187 @@
+			state.nested = { b: 222 };
+			expect( spy1 ).toHaveBeenCalledTimes( 2 );
+			state.nested.b = 2222;
+			expect( spy1 ).toHaveBeenCalledTimes( 2 );
+		} );
+
+		it( 'should support returning peek from getters', () => {
+			const state = proxifyState( 'test', {
+				counter: 1,
+				get double() {
+					return state.counter * 2;
+				},
+			} );
+			expect( peek( state, 'double' ) ).toBe( 2 );
+			state.counter = 2;
+			expect( peek( state, 'double' ) ).toBe( 4 );
+		} );
+
+		it( 'should support peeking getters accessing the scope', () => {
+			const state = proxifyState( 'test', {
+				get double() {
+					const { counter } = getContext< { counter: number } >();
+					return counter * 2;
+				},
+			} );
+
+			const context = proxifyState( 'test', { counter: 1 } );
+			const scope = { context: { test: context } };
+			const peekStateDouble = withScopeAndNs( scope, 'test', () =>
+				peek( state, 'double' )
+			);
+
+			const spy = jest.fn( peekStateDouble );
+			effect( spy );
+			expect( spy ).toHaveBeenCalledTimes( 1 );
+			expect( peekStateDouble() ).toBe( 2 );
+
+			context.counter = 2;
+
+			expect( spy ).toHaveBeenCalledTimes( 1 );
+			expect( peekStateDouble() ).toBe( 4 );
+		} );
+
+		it( 'should support peeking getters accessing other namespaces', () => {
+			const state2 = proxifyState( 'test2', {
+				get counter() {
+					const { counter } = getContext< { counter: number } >();
+					return counter;
+				},
+			} );
+			const context2 = proxifyState( 'test-2', { counter: 1 } );
+
+			const state1 = proxifyState( 'test1', {
+				get double() {
+					return state2.counter * 2;
+				},
+			} );
+
+			const peekStateDouble = withScopeAndNs(
+				{ context: { test2: context2 } },
+				'test2',
+				() => peek( state1, 'double' )
+			);
+
+			const spy = jest.fn( peekStateDouble );
+			effect( spy );
+			expect( spy ).toHaveBeenCalledTimes( 1 );
+			expect( peekStateDouble() ).toBe( 2 );
+
+			context2.counter = 2;
+
+			expect( spy ).toHaveBeenCalledTimes( 1 );
+			expect( peekStateDouble() ).toBe( 4 );
+		} );
+	} );
+
+	describe( 'refs', () => {
+		it( 'should preserve object references', () => {
+			expect( state.nested ).toBe( state.array[ 1 ] );
+
+			state.nested.b = 22;
+
+			expect( state.nested ).toBe( state.array[ 1 ] );
+			expect( state.nested.b ).toBe( 22 );
+			expect(
+				typeof state.array[ 1 ] === 'object' && state.array[ 1 ].b
+			).toBe( 22 );
+
+			state.nested = { b: 222 };
+
+			expect( state.nested ).not.toBe( state.array[ 1 ] );
+			expect( state.nested.b ).toBe( 222 );
+			expect(
+				typeof state.array[ 1 ] === 'object' && state.array[ 1 ].b
+			).toBe( 22 );
+		} );
+
+		it( 'should return the same proxy if initialized more than once', () => {
+			const raw = {};
+			const state1 = proxifyState( 'test', raw );
+			const state2 = proxifyState( 'test', raw );
+			expect( state1 ).toBe( state2 );
+		} );
+
+		it( 'should return the same proxy when trying to re-proxify a state object', () => {
```

Comment:
This?

```suggestion
		it( 'should throw when trying to re-proxify a state object', () => {
```

Reply:
Nice catch. 😄 Fixed in https://github.com/WordPress/gutenberg/pull/62734/commits/417a0c591c2ce0b57bbc0fc83fa2c5529404d9fa.